### PR TITLE
MTDSA-15011: Add new "INCORRECT_GOV_TEST_SCENARIO" error

### PR DESCRIPTION
**Pull request by sean-flynn-HMRC**
Please approve but **do NOT merge** yet. 

This PR also includes MTDSA-15012. 

In addition, this PR goes beyond these two tickets: this PR adds "RULE_INCORRECT_GOV_TEST_SCENARIO" to all 24 endpoints in v2.

### CODE REVIEW

File: app/v2/support/DownstreamResponseMappingSupport.scala
Comment (jeremystone) [inline app/v2/support/DownstreamResponseMappingSupport.scala:35]: Could probably do with a test in `DownstreamResponseMappingSupportSpec` to cover this.

**Review by jeremystone — COMMENTED** [review]
Not a full review. Just one comment plus we should put something in the RAML/YAML about this error.

File: app/v2/models/errors/mtdErrors.scala
Comment (rhysconnor) [inline app/v2/models/errors/mtdErrors.scala:139]: Could we change the format to match the other errors?

File: app/v2/support/DownstreamResponseMappingSupport.scala
Comment (sean-flynn-HMRC) [reply app/v2/support/DownstreamResponseMappingSupport.scala:35]: I believe that's already covered [here](https://github.com/hmrc/property-business-api/pull/361/files#diff-57330e552c06302dea0188f3c119e626d951939bf8201eac35ac94a3afd240a3R62)?

File: app/v2/models/errors/mtdErrors.scala
Comment (sean-flynn-HMRC) [reply app/v2/models/errors/mtdErrors.scala:135]: Yes we can

File: app/v2/support/DownstreamResponseMappingSupport.scala
Comment (jeremystone) [reply app/v2/support/DownstreamResponseMappingSupport.scala:35]: So it is! Cool.

**Review by rhysconnor — APPROVED** [review]
LGTM 🚀  (waiting for confirmation before merging)

**Pull request merged**
